bundles/org.dataflowanalysis.converter/src/org/dataflowanalysis/converter/PCMConverter.java

```diff
@@ -342,21 +428,424 @@ private Label getOrCreateDFDLabel(CharacteristicValue charValue) {
 
         return label;
     }
+    
+    /**
+     * Get or create a Label with the supplied String name and LabelType
+     * @param labelName the name of the Label
+     * @param type the LabelType
+     * @return the Label
+     */
+    private Label getOrCreateDFDLabel(String labelName, LabelType type) {
+        Label label = type.getLabel()
+                .stream()
+                .filter(f -> f.getEntityName()
+                        .equals(labelName))
+                .findFirst()
+                .orElseGet(() -> createLabel(labelName, type));
 
+        return label;
+    }
+
+    /**
+     * Creates a Label from the supplied CharacteristicValue and adds it to the LabelType
+     * @param charValue the CharacteristicValue
+     * @param type the LabelType
+     * @return the created Label
+     */
     private Label createLabel(CharacteristicValue charValue, LabelType type) {
         Label label = datadictionaryFactory.eINSTANCE.createLabel();
         label.setEntityName(charValue.getValueName());
         type.getLabel()
                 .add(label);
         return label;
     }
+    
+    /**
+     * Creates a Label with the supplied name and adds it to the LabelType
+     * @param name the name of the Label
+     * @param type the LabelType
+     * @return the created Label
+     */
+    private Label createLabel(String name, LabelType type) {
+        Label label = datadictionaryFactory.eINSTANCE.createLabel();
+        label.setEntityName(name);
+        type.getLabel()
+                .add(label);
+        return label;
+    }
 
+    /**
+     * Creates a LabelType from the supplied CharacteristicValue
+     * @param charValue the CharacteristicValue
+     * @return the created LabelType
+     */
     private LabelType createLabelType(CharacteristicValue charValue) {
         LabelType type = datadictionaryFactory.eINSTANCE.createLabelType();
         type.setEntityName(charValue.getTypeName());
         this.dataDictionary.getLabelTypes()
                 .add(type);
         return type;
     }
+    
+    /**
+     * Creates a LabelType with the supplied name
+     * @param name the name of the LabelType
+     * @return the created LabelType
+     */
+    private LabelType createLabelType(String name) {
```

Comment:
Why is this second createLabelType function needed? It has the exact same functionality as the one above

Reply:
This creates a label type, think `Location` or `Sensitivity`, whilst the other creates a label of a label type, think `EU` or `Sensitive`

Reply:
No, i mean there is createLabelType(String) and createLabelType(CharactersiticValue) where the second does the exact same as the first using the chars entity name

Reply:
Fixed in 9161111817fd